### PR TITLE
Implement an NSO loader 

### PR DESCRIPTION
**Pull request by bylaws**
This reworks the executable loading code to be common between loaders and implements an NSO loader.

Huge thanks to cohaereo for the initial implementation.

### CODE REVIEW

File: app/src/main/cpp/skyline/loader/nro.cpp [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #include <kernel/memory.h>
 #include <vfs/nacp.h>
 #include <vfs/region_backing.h>
+#include "executable.h"
```
Comment (PixelyIon) [inline app/src/main/cpp/skyline/loader/nro.cpp:9]: This include is redundant as `executable.h` is included in `loader.h` which is included here via `nro.h`

File: app/src/main/cpp/skyline/loader/nro.cpp [hunk 2]
```diff
@@ -38,52 +39,28 @@ namespace skyline::loader {
         return buffer;
     }
 
-    std::vector<u8> NroLoader::GetSegment(NroSegmentType segment) {
-        NroSegmentHeader &segmentHeader = header.segments[static_cast<int>(segment)];
-        std::vector<u8> buffer(segmentHeader.size);
+    std::vector<u8> NroLoader::GetSegment(NroSegmentHeader &segment) {
```
Comment (PixelyIon) [inline app/src/main/cpp/skyline/loader/nro.cpp:42]: The `segment` argument can be made into a `const` reference

File: app/src/main/cpp/skyline/loader/nso.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+namespace skyline::loader {
+    NsoLoader::NsoLoader(const std::shared_ptr<vfs::Backing> &backing) : Loader(backing) {
+        u32 magic{};
+        backing->Read(&magic);
+
+        if (magic != util::MakeMagic<u32>("NSO0"))
+            throw exception("Invalid NSO magic! 0x{0:X}", magic);
+    }
+
+    std::vector<u8> NsoLoader::GetSegment(const std::shared_ptr<vfs::Backing> &backing, NsoSegmentHeader &segment, u32 compressedSize) {
```
Comment (PixelyIon) [inline app/src/main/cpp/skyline/loader/nso.cpp:20]: Same as `NroLoader::GetSegment`, the `segment` argument can be made into a `const` reference

File: app/src/main/cpp/skyline/loader/nso.h [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+         */
+        struct NsoSegmentHeader {
+            u32 fileOffset; //!< The offset of the segment in the NSO
+            u32 memoryOffset; //!< The memory offset where the region should be loaded
+            u32 decompressedSize; //!< Size of the region after decompression
+        };
+        static_assert(sizeof(NsoSegmentHeader) == 0xc);
+
+        /**
+         * @brief This holds the header of an NSO file
```
Comment (PixelyIon) [inline app/src/main/cpp/skyline/loader/nso.h:39]: "This structure holds..." 😉

Comment (bylaws) [reply app/src/main/cpp/skyline/loader/nso.h:39]: There are about a million other comments that sat 'this holds'

**Pull request merged**